libs/s25main/GamePlayer.h

```diff
@@ -198,7 +198,7 @@ class GamePlayer : public GamePlayerInfo
     /// Are these players allied? (-> Teamview, attack support, ...)
     bool IsAlly(unsigned char playerId) const;
     /// Truppen bestellen
-    void OrderTroops(nobMilitary* goal, unsigned count, bool ignoresettingsendweakfirst = false) const;
+    void OrderTroops(nobMilitary* goal, std::array<unsigned, NUM_SOLDIER_RANKS> counts, unsigned max) const;
```

Comment:
What is the purpose of `max`? Can you enhance the docstring clarifying what this is supposed to do exactly, e.g. order the given amount of (additional) soldiers and what that `max` is, please